automation_tools/__init__.py

```diff
@@ -1761,15 +1760,13 @@ def copy_ssh_key(from_host, to_host):
                         '-f ~/.ssh/id_rsa -t rsa -N \'\''), host=from_host)
     if int(execute(lambda: run('[ -f ~/.ssh/id_rsa.pub ]; '
                                'echo $?'), host=from_host)[from_host]) == 0:
-        tmp_path = tempfile.mkstemp()[1]
-        local("scp -o 'StrictHostKeyChecking no' "
-              "root@{0}:~/.ssh/id_rsa.pub {1}".format(from_host, tmp_path))
+        pub_key = execute(lambda: run('cat ~/.ssh/id_rsa.pub'),
+                          host=from_host)[from_host]
         execute(lambda: run('[ ! -f ~/.ssh/authorized_keys ] && '
                             'touch ~/.ssh/authorized_keys',
                             warn_only=True), host=to_host)
-        local("cat {0} | ssh -o 'StrictHostKeyChecking no' root@{1} "
-              "'cat >> .ssh/authorized_keys'".format(tmp_path, to_host))
-        os.remove(tmp_path)
+        execute(lambda: run("echo '{0}' >> ~/.ssh/authorized_keys"
```

Comment:
The below looks better and should fit the line length. Also use single quotes for python strings and prefer double for bash commands.

```python
execute(lambda: run(
    'echo "{0}" >> ~/.ssh/authorized_keys'.format(pub_key)), host=to_host)
```